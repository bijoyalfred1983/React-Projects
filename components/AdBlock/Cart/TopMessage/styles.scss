@import '../../../theme/variables';

.errorPart {
  width: 100%;
  min-height: 50px;
  margin-top: 20px;
  padding: 15px;
  background-color: #fff;
  box-shadow: 0 0 2px 0 rgba(0, 0, 0, 0.14), 0 2px 2px 0 rgba(0, 0, 0, 0.12), 0 1px 3px 0 rgba(0, 0, 0, 0.2), inset 4px 0 0 0 #d52b1e;

  img {
    vertical-align: middle;
    padding-right: 10px;
  }
}

.successPart {
  width: 100%;
  min-height: 50px;
  margin-top: 20px;
  padding: 15px;
  background-color: #fff;
  box-shadow: 0 0 2px 0 rgba(0, 0, 0, 0.14), 0 2px 2px 0 rgba(0, 0, 0, 0.12), 0 1px 3px 0 rgba(0, 0, 0, 0.2), inset 4px 0 0 0 #237804;

  img {
    vertical-align: middle;
    padding-right: 10px;
  }
}
