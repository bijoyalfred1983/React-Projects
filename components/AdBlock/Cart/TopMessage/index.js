import React, { Component } from 'react';
import { connect } from 'react-redux';
import type { Connector } from 'react-redux';
import { removePromo } from '../../../actions/itemQuantity';
import type { Dispatch } from '../../../types';
import styles from './styles.scss';
import { jsTrack } from '../../../containers/Analytics/ClickTrack';

type Props = {
  data: any,
  flag: any,
  orderSummaryData: Object,
  removePromo: () => void,
  paymentMethod: String
  // toggeleTopMessage: boolean => void
};

type State = {
  show: boolean
};

// const TopMessage = (props: Props) => {
//   if (props.flag === 'SUCCESS') {
//     return (
//       <div>
//         <div className={styles.successPart}>
//           <img src="/images/itemSuccess-mobile.png" alt="Error icon" />
//           <span>{props.data}</span>
//         </div>
//       </div>
//     );
//   }
//   return (
//     <div>
//       <div className={styles.errorPart}>
//         <img src="/images/itemError-mobile.png" alt="Error icon" />
//         <span>{props.data}</span>
//       </div>
//     </div>
//   );
// };

class TopMessage extends Component<Props, State> {
  componentWillUnmount() {
    this.props.removePromo();
  }

  showMessage = () => {
    /* Analytics start */
    const propsData = this.props;
    const summaryData =
      propsData.orderSummaryData !== undefined
        ? this.props.orderSummaryData
        : '';
    let savings = 0;
    if (summaryData !== undefined && summaryData !== '') {
      savings = summaryData.promotionSavings
        ? Math.abs(summaryData.promotionSavings)
        : 0;
    }
    if (!global.utagData.topmsg) {
      if (propsData.flag === 'SUCCESS' && savings > 0) {
        const code = summaryData.couponVOList.couponList[0].name;
        jsTrack.successPROMOLink(code, savings);
        global.utagData.topmsg = true;
      } else if (propsData.flag === 'ERROR') {
        const errosMsg = propsData.data;
        const promocode = global.utagData.promo;
        if (promocode !== undefined) {
          jsTrack.invalidPROMOLink(promocode, errosMsg);
          jsTrack.cartItemLevelError(errosMsg);
        }
        global.utagData.topmsg = true;
      }
    }
    // Checkout Errors
    if (this.props.paymentMethod) {
      jsTrack.trackPaymentFailureCheckout(
        propsData.data,
        this.props.paymentMethod
      );
    }
    /* Analytics end */

    if (this.props.flag === 'SUCCESS') {
      return (
        <div>
          <div className={styles.successPart}>
            <img src="/images/itemSuccess-mobile.png" alt="Error icon" />
            <span>{this.props.data}</span>
          </div>
        </div>
      );
    }
    return (
      <div>
        <div className={styles.errorPart}>
          <img src="/images/itemError-mobile.png" alt="Error icon" />
          <span>{this.props.data}</span>
        </div>
      </div>
    );
  };

  render() {
    return <div>{this.showMessage()}</div>;
  }
}

/* istanbul ignore next */
const connector: Connector<{}, Props> = connect(null, (dispatch: Dispatch) => ({
  removePromo: () => dispatch(removePromo())
}));

export default connector(TopMessage);
