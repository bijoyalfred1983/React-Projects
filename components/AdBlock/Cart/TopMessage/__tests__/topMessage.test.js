import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import TopMessage from '../index';

describe('<TopMessage />', () => {
  let PROPS;
  beforeEach(() => {
    const orderSummaryData = {
      totalSavings: '20',
      couponVOList: {
        couponList: [
            {
              name: 'test'
            }
          ]
      }
    }
    PROPS = {
      data: [1, 2, 3],
      orderSummaryData
    };
  });

  test('renders on desktop/mobile correctly', () => {
     global.utagData = {
    topmsg:false
  }
    const wrapper = mount(<TopMessage {...PROPS} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('renders on desktop/mobile correctly when flag = ERROR', () => {
     global.utagData = {
    topmsg:false
  }
    PROPS.flag = 'ERROR';
    const wrapper = mount(<TopMessage {...PROPS} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('renders on desktop/mobile correctly when flag = SUCCESS', () => {
     global.utagData = {
  }
    PROPS.flag = 'SUCCESS';
    PROPS.orderSummaryData.totalSavings = 20;
    const wrapper = mount(<TopMessage {...PROPS} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
