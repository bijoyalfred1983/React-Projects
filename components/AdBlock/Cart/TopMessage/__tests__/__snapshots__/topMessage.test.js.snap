// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopMessage /> renders on desktop/mobile correctly 1`] = `
<TopMessage
  data={
    Array [
      1,
      2,
      3,
    ]
  }
>
  <div>
    <div
      className="errorPart"
    >
      1
      2
      3
    </div>
  </div>
</TopMessage>
`;

exports[`<TopMessage /> renders on desktop/mobile correctly when flag = ERROR 1`] = `
<TopMessage
  data={
    Array [
      1,
      2,
      3,
    ]
  }
  flag="ERROR"
>
  <div>
    <div
      className="errorPart"
    >
      <img
        alt="Error icon"
        src="/images/itemError-mobile.png"
      />
      <span>
        1
        2
        3
      </span>
    </div>
  </div>
</TopMessage>
`;

exports[`<TopMessage /> renders on desktop/mobile correctly when flag = SUCCESS 1`] = `
<TopMessage
  data={
    Array [
      1,
      2,
      3,
    ]
  }
  flag="SUCCESS"
>
  <div>
    <div
      className="successPart"
    >
      <img
        alt="Error icon"
        src="/images/itemSuccess-mobile.png"
      />
      <span>
        1
        2
        3
      </span>
    </div>
  </div>
</TopMessage>
`;
